labman/db/process.py

```diff
@@ -1567,6 +1554,102 @@ def concentrations(self):
                 (composition_module.Composition.factory(comp_id), r_con, c_con)
                 for comp_id, r_con, c_con in TRN.execute_fetchindex()]
 
+    def compute_concentrations(self, dna_amount=240, min_val=1, max_val=15,
+                               blanks=2, size=500):
+        """Compute the normalized concentrations
+
+        Parameters
+        ----------
+        dna_amount: float, optional
+            (Amplicon) Total amount of DNA, in ng. Default: 240
+        min_val: float, optional
+            (Amplicon) Minimum amount of DNA to normalize to (nM). Default: 1
+        max_val: float, optional
+            (Amplicon) Maximum value. Wells above this number will be
+            excluded (nM). Default: 15
+        blanks: float, optional
+            (Amplicon) Amount to pool for the blanks (nM). Default: 2.
+        size: int, optional
+            (Shotgun) The average library molecule size, in bp.
+        """
+        concentrations = self.concentrations
+        layout = concentrations[0][0].container.plate.layout
+
+        res = None
+        if isinstance(concentrations[0][0],
+                      composition_module.LibraryPrep16SComposition):
+            # Amplicon
+            sample_concs = np.zeros_like(layout, dtype=float)
+            is_blank = np.zeros_like(layout, dtype=bool)
+            for comp, r_conc, _ in concentrations:
+                well = comp.container
+                row = well.row - 1
+                col = well.column - 1
+                sample_concs[row][col] = r_conc
+                sc = comp.gdna_composition.sample_composition
+                is_blank[row][col] = sc.sample_composition_type == 'blank'
+
+            res = QuantificationProcess._compute_amplicon_pico_concentration(
+                sample_concs, dna_amount)
+            res[is_blank] = blanks
+            res[sample_concs < min_val] = min_val
```

Comment:
It doesn't seem appropriate to me to store incorrect values under a 'quantification process.' Instead, the best estimate of values should be stored, and pooling handled depending on those values.